### PR TITLE
Fixed errcheck issues in server/channels/app/export_test.go

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28922
Jira https://mattermost.atlassian.net/browse/MM-61246
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/export_test.go [hunk 1]
```diff
@@ -240,17 +244,18 @@ func TestExportDMChannel(t *testing.T) {
...
 				Category: model.PreferenceCategoryFavoriteChannel,
 				Name:     ch.Id,
 				Value:    "true",
 			},
 		})
+		require.NoError(t, err)
 
 		var b bytes.Buffer
-		err := th1.App.BulkExport(th1.Context, &b, "somePath", nil, model.BulkExportOpts{})
+		err = th1.App.BulkExport(th1.Context, &b, "somePath", nil, model.BulkExportOpts{})
```
Comment (hanzei) [inline server/channels/app/export_test.go:258]: `err` is now an `error`, but `th1.App.BulkExport` returns a `*model.AppError`. Please use a new variable.

**Review by hanzei — CHANGES_REQUESTED** [review]
LGTM, one blocking comment

**Comment by Aryakoste** [comment]
@hanzei Changes done.

File: server/channels/app/export_test.go [hunk 2]
```diff
@@ -294,8 +299,10 @@ func TestExportDMChannel(t *testing.T) {
...
 
-		th1.App.PermanentDeleteUser(th1.Context, th1.BasicUser2)
-		th1.App.PermanentDeleteUser(th1.Context, th1.BasicUser)
+		appErr := th1.App.PermanentDeleteUser(th1.Context, th1.BasicUser2)
+		require.Nil(t, appErr)
+		appErr = th1.App.PermanentDeleteUser(th1.Context, th1.BasicUser)
+		require.Nil(t, appErr)
 
 		var b bytes.Buffer
 		err := th1.App.BulkExport(th1.Context, &b, "somePath", nil, model.BulkExportOpts{})
```
Comment (hanzei) [inline server/channels/app/export_test.go:308]: This need to become an `appErr`

**Comment by Aryakoste** [comment]
@hanzei Changes done

**Review by hanzei — CHANGES_REQUESTED** [review]
You are reusing an `err` variable in L309:

```
Error: channels/app/export_test.go:309:3: calling require.Nil on error, please use require.NoError instead
```
Let's make CI happy :smile:

**Comment by Aryakoste** [comment]
@hanzei In my local branch i was not seeing this error. But when i merged the branch new code was added which is giving this error. I will fix this.

**Review by davidkrauser — APPROVED** [review]
LGTM :+1:

**Pull request merged**